app/controllers/sites_controller.rb

```diff
@@ -106,6 +106,7 @@ def users
     return unless authorize_resource(@site, READ_SITE)
 
     @users = ComputedPolicy.authorized_users(Policy::Actions::READ_SITE, @site)
+    check_access(User.within(@navigation_context.entity), READ_SITE)
```

Comment:
`check_access` will return the list of resources for which the current user can perform the chosen action. So this should be something along the lines of:
```ruby
@users = check_access(@users, READ_USER)
```

Which will keep in `@users` all the users that can be READ by the currently logged in user.